assets/src/edit-story/components/panels/border/panel.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback } from 'react';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { SimplePanel } from '../panel';
+import { Color, Numeric, Row } from '../../form';
+import { getCommonValue, useCommonColorValue } from '../utils';
+import BorderWidthControls from './borderWidth';
+
+// @todo Only display for square shapes.
+
+const BoxedNumeric = styled(Numeric)`
+  padding: 6px 6px;
+  border-radius: 4px;
+`;
+
+const Space = styled.div`
+  flex: 0 0 10px;
+`;
+
+function BorderStylePanel(props) {
+  const { selectedElements, pushUpdate } = props;
+  const borderColor = useCommonColorValue(selectedElements, 'borderColor');
+  const borderDash = getCommonValue(selectedElements, 'borderDash', 0);
+  const borderGap = getCommonValue(selectedElements, 'borderGap', 0);
+
+  const handleColorChange = useCallback(
+    (value) => pushUpdate({ borderColor: value }, true),
+    [pushUpdate]
+  );
+
+  const handleStyleChange = useCallback(
+    (value, name) => pushUpdate({ [name]: value }, true),
+    [pushUpdate]
+  );
+
+  return (
+    <SimplePanel name="borderStyle" title={__('Border', 'web-stories')}>
+      <BorderWidthControls {...props} />
+      <Row>
+        <Color
+          value={borderColor}
+          onChange={handleColorChange}
+          label={__('Border color', 'web-stories')}
+          labelId="border-color-label"
+        />
+      </Row>
+      <Row>
+        <BoxedNumeric
```

Comment:
Perhaps Gap and Dash should be together in one object.

Reply:
I feel like they should all belong to the same `border` object - so it has `{left, right, top, bottom, color, dash, gap}`. But I'm open to other variants.

Reply:
Hmm, perhaps. I actually started with all together but since `locked` is also part of it then it didn't seem clear what the `locked` would be for. In theory, the dash and gap could also be allowed to be locked in the future 🤷 We could preemptively name the `locked` to `widthLock` though so that if that should happen, `dashLock` or sth similar could easily be added instead of having to migrate. Then we could have all together in one object. Thoughts?

Reply:
Combined all together with stating that the lock is for width specifically.